File: back/Versao_Modelo.py
class Versao_Modelo:
    def __init__(self, db):
        self.db = db

    def add_versao_modelo(self, nome, status):
        try:
            self.db.cursor.execute("INSERT INTO versao_modelo (Nome, Status) VALUES (%s, %s)", (nome, status))
            self.db.conn.commit()
        except Exception as e:
            print(f"Erro ao adicionar versao do modelo ao banco de dados: {e}")
            raise e

    def get_versao_modelo(self):
        query = "SELECT * FROM processo ORDER BY id"
        print(f"Executando query: {query}")
        self.db.cursor.execute(query)
        result = self.db.cursor.fetchall()
        return result

    def delete_versao_modelo(self, versao_modelo_id):
        try:
            self.db.cursor.execute("DELETE FROM versao_modelo WHERE ID = %s", (versao_modelo_id,))
            self.db.conn.commit()
        except Exception as e:
            print(f"Erro ao deletar versao_modelo: {e}")
            raise e

    def update_versao_modelo(self, nome, status, id):
        self.db.cursor.execute("UPDATE versao_modelo SET Nome = %s, Status = %s WHERE ID = %s", (nome, status, id))
        self.db.conn.commit()

File: backjs/classes/VersaoModelo.js
class VersaoModelo {
    constructor(db) {
        this.db = db;
    }

    async addVersaoModelo(nome, status) {
        const query = "INSERT INTO versao_modelo (Nome, Status) VALUES (?, ?)";
        const values = [nome, status];
        try {
            await this.db.executeQuery(query, values);
        } catch (err) {
            console.error("Erro ao adicionar versão do modelo ao banco de dados:", err);
            throw err;
        }
    }

    async getVersaoModelo() {
        const query = "SELECT * FROM versao_modelo ORDER BY Nome DESC";
        try {
            const result = await this.db.fetchAll(query);
            return result;
        } catch (err) {
            console.error("Erro ao obter versões do modelo:", err);
            throw err;
        }
    }

    async deleteVersaoModelo(versaoModeloId) {
        const query = "DELETE FROM versao_modelo WHERE ID = ?";
        try {
            await this.db.executeQuery(query, [versaoModeloId]);
        } catch (err) {
            console.error("Erro ao deletar versão do modelo:", err);
            throw err;
        }
    }

    async updateVersaoModelo(nome, status, id) {
        const query = "UPDATE versao_modelo SET Nome = ?, Status = ? WHERE ID = ?";
        const values = [nome, status, id];
        try {
            await this.db.executeQuery(query, values);
        } catch (err) {
            console.error("Erro ao atualizar versão do modelo:", err);
            throw err;
        }
    }
}

module.exports = VersaoModelo;